From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kanaka Raju Nayana <kanaka.raju.nayana@intel.com>
Date: Wed, 29 Oct 2025 19:52:07 +0530
Subject: [PATCH] autotools/m4: manage MODULE_IMPORT_NS prototype change

Conditionally add INTEL_PMT namespace import and fix
EXPORT_SYMBOL_NS_GPL double-quoting in generated .mod.c files.

Uses feature detection instead of macro overrides to avoid interfering
with kernel's existing namespace handling.

Reference:
    cdd30ebb1b9f
    module: Convert symbol namespace to string literal

Signed-off-by: Kanaka Raju Nayana <kanaka.raju.nayana@intel.com>
---
 drivers/platform/x86/intel/pmt/crashlog.c  | 4 ++++
 drivers/platform/x86/intel/pmt/telemetry.c | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/drivers/platform/x86/intel/pmt/crashlog.c b/drivers/platform/x86/intel/pmt/crashlog.c
index 58118ef..83e9350 100644
--- a/drivers/platform/x86/intel/pmt/crashlog.c
+++ b/drivers/platform/x86/intel/pmt/crashlog.c
@@ -613,4 +613,8 @@ BACKPORT_MOD_VERSIONS
 MODULE_AUTHOR("Alexander Duyck <alexander.h.duyck@linux.intel.com>");
 MODULE_DESCRIPTION("Intel PMT Crashlog driver");
 MODULE_LICENSE("GPL v2");
+#ifdef BPM_MODULE_IMPORT_TO_STRING_LITERAL_PRESENT
+MODULE_IMPORT_NS(INTEL_PMT);
+#else
 MODULE_IMPORT_NS("INTEL_PMT");
+#endif
diff --git a/drivers/platform/x86/intel/pmt/telemetry.c b/drivers/platform/x86/intel/pmt/telemetry.c
index 71a2214..7c5e0f4 100644
--- a/drivers/platform/x86/intel/pmt/telemetry.c
+++ b/drivers/platform/x86/intel/pmt/telemetry.c
@@ -349,4 +349,8 @@ BACKPORT_MOD_VERSIONS
 MODULE_AUTHOR("David E. Box <david.e.box@linux.intel.com>");
 MODULE_DESCRIPTION("Intel PMT Telemetry driver");
 MODULE_LICENSE("GPL v2");
+#ifdef BPM_MODULE_IMPORT_TO_STRING_LITERAL_PRESENT
+MODULE_IMPORT_NS(INTEL_PMT);
+#else
 MODULE_IMPORT_NS("INTEL_PMT");
+#endif
-- 
2.43.0

